mitmproxy/optmanager.py

```diff
@@ -612,3 +617,15 @@ def save(opts: OptManager, path: str, defaults: bool = False) -> None:
 
     with open(path, "w", encoding="utf8") as f:
         serialize(opts, f, data, defaults)
+
+
+def compute_path(script_path: str, config_path: str) -> str:
+    """
+    Make relative paths found in config files relative to said config file,
+    instead of relative to where the command is ran.
+    """
+    if os.path.isabs(script_path):
+        return script_path
+    if script_path != os.path.expanduser(script_path):
+        return script_path
+    return os.path.abspath(os.path.join(os.path.dirname(config_path), script_path))
```

Comment:
I'd like to use pathlib as much as possible, even if that comes at the cost of additional Path/str wrapping above. :) We can also make this simpler by making use of the fact that `Path("/foo") / Path("/bar")` resolves to `/bar`. :)

```suggestion
def script_path(script_path: Path, *, relative_to: Path) -> Path:
    """
    Make relative paths found in config files relative to said config file,
    instead of relative to where the command is ran.
    """
    return relative_to / script_path.expanduser().absolute()
```
(untested - relative_to requires a `relative_to or Path.cwd()` at the call site)

Reply:
I'll get on it right away.
Just a quick question, why make `relative_to` a kw_only argument here ?

Reply:
I had to handle the case where `$HOME` is not set to an absolute path (which is the case in the tox test env).